memfs/memory.go

```diff
@@ -24,11 +22,25 @@ const separator = filepath.Separator
 // Memory a very convenient filesystem based on memory files.
 type Memory struct {
 	s *storage
+
+	m mutex
```

Comment:
Why not `sync.RWMutext`?
```suggestion
	m sync.RWMutext
```